base/strings/basic.jl

```diff
@@ -613,6 +613,33 @@ isascii(c::Char) = bswap(reinterpret(UInt32, c)) < 0x80
 isascii(s::AbstractString) = all(isascii, s)
 isascii(c::AbstractChar) = UInt32(c) < 0x80
 
+@inline function _isascii(code_units::AbstractVector{CU}, first, last) where {CU}
+    r = zero(CU)
+    for n = first:last
+        @inbounds r |= code_units[n]
+    end
+    return 0 ≤ r < 0x80
+end
+
+@inline _isascii(::Val{N},cu::AbstractVector{CU}, first) where {N,CU} = @inline _isascii(cu,first,first+N-1)
+
+#The chunking algorithm makes the last two chunks overlap inorder to keep the size fixed
+@inline function  _isascii_chunks(::Val{N},cu::AbstractVector{CU}, first,last) where {N,CU}
+    for n=first:N:last - N
+        _isascii(Val(N),cu,n) || return false
+   end
+   return  _isascii(Val(N),cu,last-N+1)
+end
+
+function isascii(cu::AbstractVector{CU}) where {CU <: Integer}
+    chunk_size = 1024
+    chunk_threshold =  chunk_size + (chunk_size ÷ 2)
+    l = length(cu)
+    l < chunk_threshold && return _isascii(cu,1,l)
+    # return _isascii_chunks(chunk_size,cu,1,l)
+    return _isascii_chunks(Val(chunk_size),cu,1,l)
```

Comment:
Does `Val` actually make a difference here? Since `chunk_size` is already a constant, I'd expect constant propagation to already do all of the work required. Dynamically creating it here is technically type unstable, the dynamic dispatch of which is only prevented due to constant propagation already fixing the type in place.

Reply:
The Val benefit is within the noise, but the hope there is that it is the added push needed to make sure the best optimization always happens.

Reply:
You can check with `@code_warntype` or JET.jl or Cthulhu.jl to see if dynamic dispatch there actually occurs - I doubt it, in which case the `Val` is just added noise/more type inference to run.